src/main/scala/firrtl/transforms/RemoveReset.scala

```diff
@@ -77,12 +78,19 @@ object RemoveReset extends Transform with DependencyAPIMigration {
           // Add register reset to map
           resets(rname) = Reset(reset, init, info)
           reg.copy(reset = Utils.zero, init = WRef(reg))
+        case reg @ DefRegister(info, rname, _, _, reset, init) if reset.tpe == AsyncResetType =>
+          asyncResets(rname) = Reset(reset, init, info)
+          reg
         case Connect(info, ref @ WRef(rname, _, RegKind, _), expr) if resets.contains(rname) =>
           val reset = resets(rname)
           val muxType = Utils.mux_type_and_widths(reset.value, expr)
           // Use reg source locator for mux enable and true value since that's where they're defined
           val infox = MultiInfo(reset.info, reset.info, info)
           Connect(infox, ref, Mux(reset.cond, reset.value, expr, muxType))
+        case Connect(info, ref @ WRef(rname, _, RegKind, _), expr) if asyncResets.contains(rname) =>
+          val reset = asyncResets(rname)
+          val infox = MultiInfo(reset.info, reset.info, info)
+          Connect(infox, ref, expr)
```

Comment:
This need some comments to explain why `muxType` is in sync reset register while not in async. This is confusing, and may blame to Emitter handled this:
https://github.com/chipsalliance/firrtl/blob/dc0037e2c87d04019d8a705f49ca40128e48bc49/src/main/scala/firrtl/backends/verilog/VerilogEmitter.scala#L795

Reply:
https://github.com/chipsalliance/firrtl/pull/2451/commits/a82aa44b15d8588a3e4b1135a096cb7371bcc30d